src/components/d4-module/types.ts

```diff
@@ -1,70 +1,67 @@
 // Divi dependencies.
 import { ModuleEditProps } from '@divi/module-library';
 import {
-  AdminLabelGroupAttr,
-  AnimationGroupAttr,
-  BackgroundGroupAttr,
-  BorderGroupAttr,
-  BoxShadowGroupAttr,
   CssAttr,
-  DisabledOnGroupAttr,
-  FiltersGroupAttr,
-  FontBodyGroupAttr,
-  FontGroupAttr,
-  FormatAttr,
+  FormatBreakpointStateAttr,
   IdClassesGroupAttr,
   InternalAttrs,
   LinkGroupAttr,
-  OverflowGroupAttr,
-  PositionGroupAttr,
-  ScrollGroupAttr,
-  SizingGroupAttr,
-  SpacingGroupAttr,
-  StickyGroupAttr,
   TextGroupAttr,
-  TransformGroupAttr,
-  TransitionGroupAttr,
-  ZIndexGroupAttr,
+  PickedElementAttrs,
+  ModuleMetaAttrs,
+  CssGroupAttr,
 } from '@divi/types';
 
 export interface D4ModuleCssAttr extends CssAttr {
   title?: string;
   content?: string;
 }
 
-export type D4ModuleCssGroupAttr = FormatAttr<D4ModuleCssAttr>;
+export type D4ModuleCssGroupAttr = FormatBreakpointStateAttr<D4ModuleCssAttr>;
 
 export interface D4ModuleAttrs extends InternalAttrs {
+  // CSS options is used across multiple elements inside the module thus it deserves its own top property.
+  css?: CssGroupAttr;
 
-  // Content tab.
-  title?: FormatAttr<string>;
-  content?: FormatAttr<string>;
-  link?: LinkGroupAttr;
-  background?: BackgroundGroupAttr;
-  adminLabel?: AdminLabelGroupAttr;
+  module?: {
+    meta?: ModuleMetaAttrs;
+    advanced?: {
+      link?: LinkGroupAttr;
+      htmlAttributes?: IdClassesGroupAttr;
+      text?: TextGroupAttr;
+    };
+    decoration?: PickedElementAttrs<
+      'animation' |
+      'background' |
+      'border' |
+      'boxShadow' |
+      'disabledOn' |
+      'filters' |
+      'overflow' |
+      'position' |
+      'scroll' |
+      'sizing' |
+      'spacing' |
+      'sticky' |
+      'transform' |
+      'transition' |
+      'zIndex'
+    >;
+  };
+  title?: {
+    innerContent?: FormatBreakpointStateAttr<string>;
+    decoration?: PickedElementAttrs<'font'>;
+  };
+  content?: {
+    innerContent?: FormatBreakpointStateAttr<string>;
+    decoration?: PickedElementAttrs<'font'>;
+  };
 
-  // Design tab.
-  text?: TextGroupAttr;
-  titleFont?: FontGroupAttr;
-  bodyFont?: FontBodyGroupAttr;
-  sizing?: SizingGroupAttr;
-  spacing?: SpacingGroupAttr;
-  border?: BorderGroupAttr;
-  boxShadow?: BoxShadowGroupAttr;
-  filter?: FiltersGroupAttr;
-  transform?: TransformGroupAttr;
-  animation?: AnimationGroupAttr;
-
-  // Advanced tab.
-  module?: IdClassesGroupAttr;
-  css?: D4ModuleCssGroupAttr;
-  disabledOn?: DisabledOnGroupAttr;
-  overflow?: OverflowGroupAttr;
-  transition?: TransitionGroupAttr;
-  position?: PositionGroupAttr;
-  zIndex?: ZIndexGroupAttr;
-  sticky?: StickyGroupAttr;
-  scroll?: ScrollGroupAttr;
+  image?: {
```

Comment:
Is there any specific reason to add image? `d4_module` don't have any image.

Reply:
Yes, because we have an `Image` setting field in this commit: https://github.com/elegantthemes/d5-extension-example-modules/blob/01412a5689a5cbdc7fcf62355c2ae2ec3b427c8e/src/components/d4-module/settings-content.tsx#L47

Reply:
We should remove the image setting right?

Reply:
Addressed in https://github.com/elegantthemes/d5-extension-example-modules/pull/12/commits/ff7169dc17493ee49193e837e2f53adebb5d0ec3